presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -371,18 +373,20 @@ public TaskInfo updateTask(
 
         SqlTask sqlTask = tasks.getUnchecked(taskId);
         sqlTask.recordHeartbeat();
+        sqlTask.updateCommunicationSlug(communicationSlug);
```

Comment:
Probably we should just separate create task and update task? This logic can get very tricky if multiple tasks are updating it at same time.  And we do not check if somebody is trying to update the slug to another value maliciously (it will silent fail based on updateCommunicationSlug logic).